### PR TITLE
Add methods to QuicChannel that allows to query if a connection was t…

### PR DESCRIPTION
**Pull request by normanmaurer**
…imed out

Motivation:

Sometimes we would like to know if a connection was closed because of a timeout. This was not possible.

Modifications:

Add isTimedOut() method which allows to query if the connection was closed because of the idle time out

Result:

Be able to know if a connection is closed because of the idle timeout

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
     <quicheHomeIncludeDir>${quicheHomeDir}/include</quicheHomeIncludeDir>
     <quicheRepository>https://github.com/cloudflare/quiche</quicheRepository>
     <quicheBranch>master</quicheBranch>
-    <quicheCommitSha>e805323103afc584259be54d38c804dec7888bd6</quicheCommitSha>
+    <quicheCommitSha>e9f59a55f5b56999d0e609a73aa8f1b450878a0c</quicheCommitSha>
```
Comment (NiteshKant) [inline pom.xml:98]: Can we split the version upgrade to a different PR? It is easier to track.

File: src/main/java/io/netty/incubator/codec/quic/QuicChannel.java [hunk 2]
```diff
@@ -165,6 +165,13 @@ default ChannelPromise voidPromise() {
      */
     long peerAllowedStreams(QuicStreamType type);
 
+    /**
+     * Returns {@code true} if the connection was closed because of the idle timeout.
+     *
+     * @return {@code true} if the connection was closed because of the idle timeout, {@code false}.
```
Comment (NiteshKant) [inline src/main/java/io/netty/incubator/codec/quic/QuicChannel.java:171]:
```suggestion
     * Returns {@code true} if the connection was closed because of idle timeout.
     *
     * @return {@code true} if the connection was closed because of idle timeout, {@code false}.
```

File: src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java [hunk 3]
```diff
@@ -333,6 +339,7 @@ void forceClose() {
             }
             connection = null;
             state = CLOSED;
+            timedOut = Quiche.quiche_conn_is_timed_out(conn.address());
```
Comment (NiteshKant) [inline src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java:342]: Is this method guaranteed to be called for all closures? If not, should we query and cache this `boolean` on demand if the connection is closed?

```java
        final QuicheQuicConnection conn = this.connection;
        // modify `timedOut` to a `byte`
        if (conn.isClosed() && timedOut == -1) {
            timedOut = Quiche.quiche_conn_is_timed_out(conn.address()) ? (byte) 0b1 : (byte) 0;
        }
        return timedOut == 1;
```

File: pom.xml (hunk 1 above)
Comment (normanmaurer) [reply pom.xml:98]: Sure ... that said upgrade is required for this feature

File: src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java (hunk 3 above)
Comment (normanmaurer) [reply src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java:342]: yes its guaranteed.

**Review by NiteshKant — APPROVED** [review]
